control/control.go

```diff
@@ -911,8 +1003,43 @@ func (p *pluginControl) CollectMetrics(metricTypes []core.Metric, deadline time.
 
 	// For each available plugin call available plugin using RPC client and wait for response (goroutines)
 	for pluginKey, pmt := range pluginToMetricMap {
+
+		// if plugin does not have a pool, try to make new subscription
+		if !p.pluginRunner.AvailablePlugins().hasPool(pluginKey) {
+			// this happens only when the queried metric in task manifest matches with metrics
```

Comment:
@IzabellaRaulin: It's the plan to move the logic here, dealing with plugins being loaded that may match a query, to the loaded plugin event handler, right?